packages/premint-sdk/src/premint-api.ts

```diff
@@ -0,0 +1,684 @@
+import { createPublicClient, decodeEventLog, http } from "viem";
+import type {
+  Account,
+  Address,
+  Chain,
+  Hex,
+  PublicClient,
+  TransactionReceipt,
+  WalletClient,
+} from "viem";
+import {
+  zoraCreator1155PremintExecutorImplABI,
+  zoraCreator1155PremintExecutorImplAddress,
+  zoraCreatorFixedPriceSaleStrategyAddress,
+} from "@zoralabs/zora-1155-contracts";
+import { foundry, zora, zoraTestnet } from "viem/chains";
+import { PremintConfig, preminterTypedDataDefinition } from "./preminter";
+
+export type NetworkConfig = {
+  chainId: number;
+  zoraPathChainName: string;
+  zoraBackendChainName: string;
+  isTestnet: boolean;
+};
+
+export const enum BackendChainNames {
+  ZORA_MAINNET = "ZORA-MAINNET",
+  ZORA_GOERLI = "ZORA-GOERLI",
+}
+
+async function wait(delayMs: number) {
+  return new Promise((resolve) => {
+    setTimeout(resolve, delayMs);
+  });
+}
+
+const ZORA_API_BASE = "https://api.zora.co/premint/";
```

Comment:
Have we thought about how we'll handle `v1`/`v2` versions of premint? Will the API still stay the same?

Reply:
We probably will need to handle different messages coming back from the server transparently.